### PR TITLE
Play next color

### PR DESCRIPTION
**Pull request by Octoton**
Resolve https://github.com/VinylMusicPlayer/VinylMusicPlayer/issues/701

### CODE REVIEW

File: app/src/main/java/com/poupa/vinylmusicplayer/ui/fragments/player/AbsPlayerFragment.java [hunk 1]
```diff
@@ -254,4 +255,27 @@ public Callbacks getCallbacks() {
     public interface Callbacks {
         void onPaletteColorChanged();
     }
+
+    protected static int adjustLightness(int color, float lightnessFactor) {
```
Comment (soncaokim) [inline app/src/main/java/com/poupa/vinylmusicplayer/ui/fragments/player/AbsPlayerFragment.java:259]: Is it possible to move these functions to VinylMusicPlayerColorUtil class?

**Comment by soncaokim** [comment]
The dark theme rendering is not always readable - for example see screenshot
Screenshot_20230313-205135_Vinyl Music Player

**Comment by Octoton** [comment]
Tried a linear approach, seems to give better contrast when needed

**Comment by soncaokim** [comment]
Hi, while the contrast is definitely better, I still dont quite like the result (the generated color is likely far from the cover art chosen color).

This is obviously very personal / subjective --> Another opinion is needed (@AdrienPoupa ?)

File: app/src/main/java/com/poupa/vinylmusicplayer/util/VinylMusicPlayerColorUtil.java [hunk 2]
```diff
@@ -92,4 +93,32 @@ public static int getSystemNightMode(@NonNull final Context context) {
...
+        ColorUtils.colorToHSL(color, outHsl);
+
+        outHsl[2] = outHsl[2] * lightnessFactor;
+        return ColorUtils.HSLToColor(outHsl);
+    }
+
+    public static int getContrastedColor(int foreground, int background) {
+        double contrast = ColorUtils.calculateContrast(foreground, background);
+
+        if (contrast < 4.5) {
```
Comment (soncaokim) [inline app/src/main/java/com/poupa/vinylmusicplayer/util/VinylMusicPlayerColorUtil.java:108]: Can you please give `4.5f` a name, i.e `minContrast`?

File: app/src/main/java/com/poupa/vinylmusicplayer/ui/fragments/player/AbsPlayerFragment.java [hunk 3]
```diff
@@ -12,6 +12,7 @@
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.appcompat.widget.Toolbar;
+import androidx.core.graphics.ColorUtils;
```
Comment (soncaokim) [inline app/src/main/java/com/poupa/vinylmusicplayer/ui/fragments/player/AbsPlayerFragment.java:15]: Not necessary import?

**Review by soncaokim — APPROVED** [review]
Look good to me. Few minor comment left inline

**Comment by soncaokim** [comment]
@Octoton can you please have a look at my last comments. They would require minor change in the PR, so that we can merge this in?

Thanks

**Pull request merged**